### ISSUE TITLE
Update wiki around SC2213 to include examples of long option names handling with getopts

### ISSUE THREAD

**Issue by andyfeller** [koalaman/shellcheck#2351]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2213
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

LOCK_REPOSITORIES=false

while getopts "af:hlo:r:t:-:" OPT; do
  if [ "$OPT" = "-" ]; then   # long option: reformulate OPT and OPTARG
    OPT="${OPTARG%%=*}"       # extract long option name
    OPTARG="${OPTARG#$OPT}"   # extract long option argument (may be empty)
    OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
  fi

  case "$OPT" in
    lock-repositories)
      LOCK_REPOSITORIES=true
      ;;
  esac
done
```

#### Here's what shellcheck currently says:
In ... line 60:
    lock-repositories)
    ^---------------^ SC2214: This case is not specified by getopts.

#### Here's what I wanted or expected to see:

I would like to see https://github.com/koalaman/shellcheck/wiki/SC2213 containing an example of how it recommends long options being handled as it appears [`v0.7.2` release has logic around them](https://github.com/koalaman/shellcheck/commit/f100c2939e7fd50975b20d3c7e7cd9e635b1b428).  I know this is a very gray area, so I'm just focusing on how to do long option names in a way that passes shellcheck 👍

**Comment by koalaman** [comment]
`getopts` does not support long options, and this is an interesting way around that that I haven't seen before. I've updated the check to bail if it sees the `getopts` variable being modified, and added some suggestions to the wiki like using `getopt` (no "s"), or checking in a `-)` branch. Thanks!

(The `getopts` code you link to is unrelated, it's what ShellCheck uses internally for analysis purposes)
